tests_integ/helpers/base_test.py

```diff
@@ -0,0 +1,379 @@
+
+        cls.file_to_s3_uri_map = FILE_TO_S3_URI_MAP
+        cls.code_key_to_file = CODE_KEY_TO_FILE_MAP
+
+        if not cls.output_dir.exists():
+            os.mkdir(str(cls.output_dir))
+
+        cls._upload_resources(FILE_TO_S3_URI_MAP)
+
+    @classmethod
+    def tearDownClass(cls):
+        cls._clean_bucket()
+
+    @classmethod
+    def _clean_bucket(cls):
+        """
+        Empties and deletes the bucket used for the tests
+        """
+        s3 = boto3.resource("s3")
+        bucket = s3.Bucket(cls.s3_bucket_name)
+        object_summary_iterator = bucket.objects.all()
+
+        for object_summary in object_summary_iterator:
+            try:
+                cls.s3_client.delete_object(Key=object_summary.key, Bucket=cls.s3_bucket_name)
+            except ClientError as e:
+                LOG.error(
+                    "Unable to delete object %s from bucket %s", object_summary.key, cls.s3_bucket_name, exc_info=e
+                )
+        try:
+            cls.s3_client.delete_bucket(Bucket=cls.s3_bucket_name)
+        except ClientError as e:
+            LOG.error("Unable to delete bucket %s", cls.s3_bucket_name, exc_info=e)
+
+    @classmethod
+    def _upload_resources(cls, file_to_s3_uri_map):
+        """
+        Creates the bucket and uploads the files used by the tests to it
+        """
+        if not file_to_s3_uri_map or not file_to_s3_uri_map.items():
+            LOG.debug("No resources to upload")
+            return
+
+        create_bucket(cls.s3_bucket_name, region=cls.my_region)
+
+        current_file_name = ""
+
+        try:
+            for file_name, file_info in file_to_s3_uri_map.items():
+                current_file_name = file_name
+                code_path = str(Path(cls.code_dir, file_name))
+                LOG.debug("Uploading file %s to bucket %s", file_name, cls.s3_bucket_name)
+                cls.s3_client.upload_file(code_path, cls.s3_bucket_name, file_name)
+                LOG.debug("File %s uploaded successfully to bucket %s", file_name, cls.s3_bucket_name)
+                file_info["uri"] = cls._get_s3_uri(file_name, file_info["type"])
+        except ClientError as error:
+            LOG.error("Upload of file %s to bucket %s failed", current_file_name, cls.s3_bucket_name, exc_info=error)
+            cls._clean_bucket()
+            raise error
+
+    @classmethod
+    def _get_s3_uri(cls, file_name, uri_type):
+        if uri_type == "s3":
+            return "s3://{}/{}".format(cls.s3_bucket_name, file_name)
+
+        if cls.my_region == "us-east-1":
+            return "https://s3.amazonaws.com/{}/{}".format(cls.s3_bucket_name, file_name)
+        if cls.my_region == "us-iso-east-1":
+            return "https://s3.us-iso-east-1.c2s.ic.gov/{}/{}".format(cls.s3_bucket_name, file_name)
+        if cls.my_region == "us-isob-east-1":
+            return "https://s3.us-isob-east-1.sc2s.sgov.gov/{}/{}".format(cls.s3_bucket_name, file_name)
```

Comment:
Is there no way to get this from S3 itself? Seems like a maintenance burden otherwise.

Reply:
It seems you can only get a [presigned temporary URL](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.generate_presigned_url) from the client.